packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -156,10 +158,51 @@ class ReleaseUnpackMacOS extends UnpackMacOS {
   String get name => 'release_unpack_macos';
 
   @override
-  List<Source> get inputs => <Source>[
-    ...super.inputs,
-    const Source.artifact(Artifact.flutterMacOSXcframework, mode: BuildMode.release),
+  List<Source> get outputs => super.outputs + const <Source>[
+    Source.pattern('{OUTPUT_DIR}/FlutterMacOS.framework.dSYM/Contents/Resources/DWARF/FlutterMacOS'),
   ];
+
+  @override
+  List<Source> get inputs => super.inputs + const <Source>[
+    Source.artifact(Artifact.flutterMacOSXcframework, mode: BuildMode.release),
+  ];
+
+  @override
+  Future<void> build(Environment environment) async {
+    await super.build(environment);
+
+    // Copy Flutter framework dSYM (debug symbol) bundle, if present.
+    final String? buildModeEnvironment = environment.defines[kBuildMode];
```

Comment:
I'm guessing yes because it's not being copied into the app, but this is safe to do post-codesigning the framework, right?